x-pack/plugins/cross_cluster_replication/__jest__/client_integration/follower_index_edit.test.js

```diff
@@ -0,0 +1,160 @@
+
+const FOLLOWER_INDEX = {
+  name: FOLLOWER_INDEX_NAME,
+  remoteCluster: 'new-york',
+  leaderIndex: 'some-leader-test',
+  status: 'active',
+  maxReadRequestOperationCount: 7845,
+  maxOutstandingReadRequests: 16,
+  maxReadRequestSize: '64mb',
+  maxWriteRequestOperationCount: 2456,
+  maxWriteRequestSize: '1048b',
+  maxOutstandingWriteRequests: 69,
+  maxWriteBufferCount: 123456,
+  maxWriteBufferSize: '256mb',
+  maxRetryDelay: '225ms',
+  readPollTimeout: '2m'
+};
+
+const testBedOptions = {
+  memoryRouter: {
+    onRouter: (router) => routing.reactRouter = router,
+    // The auto-follow pattern id to fetch is read from the router ":id" param
+    // so we first set it in our initial entries
+    initialEntries: [`/${FOLLOWER_INDEX_NAME}`],
+    // and then we declarae the :id param on the component route path
+    componentRoutePath: '/:id'
+  }
+};
+
+describe('Edit Auto-follow pattern', () => {
+  let server;
+  let find;
+  let component;
+  let getUserActions;
+  let getFormErrorsMessages;
+  let clickSaveForm;
+  let setLoadRemoteClusteresResponse;
+  let setGetFollowerIndexResponse;
+
+  beforeEach(() => {
+    server = sinon.fakeServer.create();
+    server.respondImmediately = true;
+
+    // Register helpers to mock Http Requests
+    ({
+      setLoadRemoteClusteresResponse,
+      setGetFollowerIndexResponse
+    } = registerHttpRequestMockHelpers(server));
+
+    // Set "default" mock responses by not providing any arguments
+    setLoadRemoteClusteresResponse();
+
+    // Mock all HTTP Requests that have not been handled previously
+    server.respondWith([200, {}, '']);
+  });
+
+  describe('on component mount', () => {
+    const remoteClusters = [
+      { name: 'new-york', seeds: ['localhost:123'], isConnected: true },
+    ];
+
+    beforeEach(async () => {
+      setLoadRemoteClusteresResponse(remoteClusters);
+      setGetFollowerIndexResponse(FOLLOWER_INDEX);
+      ({ component, find } = initTestBed(FollowerIndexEdit, undefined, testBedOptions));
+
+      await nextTick();
+      component.update();
+    });
+
+    /**
+     * As the "edit" follower index component uses the same form underneath that
+     * the "create" follower index, we won't test it again but simply make sure that
+     * the form component is indeed shared between the 2 app sections.
+     */
+    test('should use the same Form component that the "<FollowerIndexAdd />" component', async () => {
+      const { component: addFollowerIndexComponent } = initTestBed(FollowerIndexAdd, undefined, testBedOptions);
+
+      await nextTick();
+      addFollowerIndexComponent.update();
+
+      const formEdit = component.find(FollowerIndexForm);
+      const formAdd = addFollowerIndexComponent.find(FollowerIndexForm);
+
+      expect(formEdit.length).toBe(1);
+      expect(formAdd.length).toBe(1);
+    });
+
+    test('should populate the form fields with the values from the follower index loaded', () => {
```

Comment:
Could we use a data structure to dedupe some of these assertions and make it easier to compare inputs to expected values?

```js
    test('should populate the form fields with the values from the follower index loaded', () => {
      const {
        remoteCluster,
        leaderIndex,
        name,
        maxReadRequestOperationCount,
        maxOutstandingReadRequests,
        maxReadRequestSize,
        maxWriteRequestOperationCount,
        maxWriteRequestSize,
        maxOutstandingWriteRequests,
        maxWriteBufferCount,
        maxWriteBufferSize,
        maxRetryDelay,
        readPollTimeout,
      } = FOLLOWER_INDEX;

      const inputToValueMap = {
        ccrRemoteClusterInput: remoteCluster,
        ccrFollowerIndexFormLeaderIndexInput: leaderIndex,
        ccrFollowerIndexFormFollowerIndexInput: name,
        ccrFollowerIndexFormMaxReadRequestOperationCountInput: maxReadRequestOperationCount,
        ccrFollowerIndexFormMaxOutstandingReadRequestsInput: maxOutstandingReadRequests,
        ccrFollowerIndexFormMaxReadRequestSizeInput: maxReadRequestSize,
        ccrFollowerIndexFormMaxWriteRequestOperationCountInput: maxWriteRequestOperationCount,
        ccrFollowerIndexFormMaxWriteRequestSizeInput: maxWriteRequestSize,
        ccrFollowerIndexFormMaxOutstandingWriteRequestsInput: maxOutstandingWriteRequests,
        ccrFollowerIndexFormMaxWriteBufferCountInput: maxWriteBufferCount,
        ccrFollowerIndexFormMaxWriteBufferSizeInput: maxWriteBufferSize,
        ccrFollowerIndexFormMaxRetryDelayInput: maxRetryDelay,
        ccrFollowerIndexFormReadPollTimeoutInput: readPollTimeout,
      };

      Object.keys(inputToValueMap).forEach(input => {
        expect(find(input).props().value).toBe(inputToValueMap[input]);
      });
    });
```

Reply:
Good idea!